examples/sugarscape_g1mt/app.py

```diff
@@ -0,0 +1,61 @@
+import numpy as np
+import solara
+from matplotlib.figure import Figure
+from mesa_models.experimental import JupyterViz
+from sugarscape_g1mt.model import SugarscapeG1mt
+from sugarscape_g1mt.resource_agents import Sugar
+from sugarscape_g1mt.trader_agents import Trader
+
+
+def space_drawer(viz):
+    def portray(g):
+        layers = {
+            "sugar": [[np.nan for j in range(g.height)] for i in range(g.width)],
+            "spice": [[np.nan for j in range(g.height)] for i in range(g.width)],
+            "trader": {"x": [], "y": [], "c": "tab:red", "marker": "o", "s": 10},
+        }
+
+        for i in range(g.width):
+            for j in range(g.height):
+                content = g._grid[i][j]
+                for agent in content:
+                    if isinstance(agent, Trader):
+                        layers["trader"]["x"].append(i)
+                        layers["trader"]["y"].append(j)
+                    else:
+                        # Don't visualize resource with value <= 1.
+                        value = agent.amount if agent.amount > 1 else np.nan
+                        if isinstance(agent, Sugar):
+                            layers["sugar"][i][j] = value
+                        else:
+                            layers["spice"][i][j] = value
+        return layers
+
+    fig = Figure()
+    ax = fig.subplots()
+    out = portray(viz.model.grid)
+    # Sugar
+    im = ax.imshow(out["sugar"], cmap="spring")
+    fig.colorbar(im, orientation="vertical")
+    # Spice
+    ax.imshow(out["spice"], cmap="winter")
+    # Trader
+    ax.scatter(**out["trader"])
+    ax.set_axis_off()
+    solara.FigureMatplotlib(fig, dependencies=[viz.model, viz.df])
+
+
+model_params = {
+    "width": 50,
+    "height": 50,
+}
+
+page = JupyterViz(
+    SugarscapeG1mt,
+    model_params,
+    measures=["Trader", "Price"],
+    name="Sugarscape {G1, M, T}",
+    space_drawer=space_drawer,
+    play_interval=1500,
```

Comment:
This has to be increased from the default 400 ms, because otherwise there is a "traffic jam" of the steps, which makes it impossible to pause the simulation immediately. The correct implementation should have been to use delay between steps instead of interval. But this is not doable via `ipywidgets.Play`. This requires the threading version instead. Solara dev is currently working to make `use_threading` work.